### ISSUE TITLE
Live Demo

### ISSUE THREAD

**Issue by codydaig** [meanjs/mean#1044]
We should get another Live Demo up and available for testing.

My proposal:
I've got a cluster of servers that I can throw the demo on that would be in sync with the master branch of the repo. I've got a handful of projects already running in this manner, and adding one more to the cluster would just be a matter of a couple minutes of configuration. I can also create a script to wipe and reseed the database every couple hours to ensure that there's always an admin and user account available for testing. 

Other Options:
- Set-up a auto deploying heroku app. This would also require a mongolab (or similar) account for the database. One downside that I see to this is that the heroku app would go to sleep and thus takes longer to load and give the user a good experience.
- Try to see if we can get free AWS credit for being an open source project to host the live demo in relatively the same form as above.
- ....? Probably plenty others, so open to discussion.

**Comment by lirantal** [comment]
To keep up with the open source model and transparency we should rely on a public service just like we do with Travis-CI, Coveralls, etc.

**Comment by codydaig** [comment]
@lirantal What kind of public service gives that level of transparency though?

**Comment by lirantal** [comment]
I was referring to the project's transparency level, meaning that both me and Ilan are able to configure Travis or Coveralls as required due to our permissions on the project and the Github integration that enables this.
I'm sure we can find some service that provides integration with Github for this use case.

**Comment by codydaig** [comment]
@lirantal The closest one that I know of would be heroku, but it doesn't integrate with github permissions, whoever creates it would have to add contributors too it manually. But I',m not sure if mongo labs does the same thing or not. If anyone has any other suggestions that integrates with github even more I'm open to those ideas!

**Comment by jloveland** [comment]
@codydaig have you tried openshift? I have not tried yet, but ...

1. it looks to be free for up to 3 nodes https://www.openshift.com/pricing/plan-comparison.html
2. you can push from travis http://docs.travis-ci.com/user/deployment/openshift/
3. it has a mean tutorial with the fullstack project https://developers.openshift.com/en/node-js-example-meanstack.html

**Comment by ilanbiala** [comment]
@jloveland I have tried openshift, but I ran into deployment issues and had trouble actually getting a live server. If you can get anything working, we'll probably look at that as it's free.

**Comment by codydaig** [comment]
@ryanjbaxter Would Bluemix have an option that would fit our needs?

**Comment by ryanjbaxter** [comment]
@codydaig potentially.  We have IBM DevOps services which is part of Bluemix and would allow you to setup a build and deploy pipeline so that when a commit is made to the repo, a build is done and deployed to bluemix.  You can choose to use the free tier MongoLabs service or a build in experimental mongodb service for the db.  (You can even setup and configure the other social services if you want)

The one downside, right now anyways, is who owns the account it gets deployed to.  You get a 30 day free trial but then someone needs to enter a credit card for the account.  However you can continue to run a single instance of a MEAN app for free even after the 30 days as it should fall below the 375 GB/hrs you get free each month after the 30 days.

**Comment by sspohC** [comment]
I'm pretty new here, but why not use AWS? Its fairly priced and a great platform

**Comment by codydaig** [comment]
@sspohC We're looking for free and AWS (at least in my experience) doesn't make permissions easy.

**Comment by sspohC** [comment]
Yeah the IAM permissions can be painful. I have associates at AWS that I would be happy to contact to see if free is feasible. I'm just getting into meanjs and I love it. Glad to help anyway I can!
